### PR TITLE
PMP: triangulate_facets_plugin for Surface_mesh

### PR DESCRIPTION
**Pull request by afabri**
Addresses Issue #1931

### CODE REVIEW

**Comment by maxGimeno** [comment]
@lrineau : This PR is based on master and is ready to be tested.

File: Polyhedron/demo/Polyhedron/Plugins/PMP/Triangulate_facets_plugin.cpp [hunk 1]
```diff
@@ -38,14 +39,16 @@ class Polyhedron_demo_triangulate_facets_plugin :
...
+      if ( qobject_cast<Scene_polyhedron_item*>(scene->item(index)) ||
+           qobject_cast<Scene_surface_mesh_item*>(scene->item(index)) )
         return true;
     }
     return false;
   }
 
 public Q_SLOTS:
    void triangulate() {
+      QApplication::setOverrideCursor(Qt::WaitCursor);
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Plugins/PMP/Triangulate_facets_plugin.cpp]: The indentation of that line is not coherent with the next line.

**Comment by lrineau** [comment]
Build errors:
```
PluginsPMP/Triangulate_facets_plugin.cpp:80:32: error: 'SMesh' is not a member of 'Scene_surface_mesh_item'
       Scene_surface_mesh_item::SMesh* pMesh = sm_item->polyhedron();
                                ^~~~~
```
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.11-Ic-52/Polyhedron_Demo/TestReport_lrineau_ArchLinux.gz

**Comment by maxGimeno** [comment]
This PR is extracted from #1961 and should be closed and ignored.

**Pull request closed without merging**